### DISCUSSION TITLE
Wine wow64 on box64 discussion

### DISCUSSION THREAD

**Discussion by agrecascino** [ptitSeb/box64#863] [General]
Earlier this year, WINE implemented a WOW64 mode which thunks 32bit calls made in wine applications to 64 bit function calls on the host operating system, analogous to WOW64 on Windows proper. For platforms with no multilib support(NVIDIA X1, X2, Xavier, Orin all lack 32 bit libgl) this will allow the emulation of 32 bit windows applications under WINE with hardware acceleration. 

As of a few days ago, box64 can now recompile 32 bit code in 64 bit programs and is able to take advantage of WINE's WOW64 mode, however it is alpha-quality. The same applies for Wine's WOW64 support itself, with some applications not working on native AMD64 either. 
If you want to try this out, you're going to want box64 git from after bb3c647f, but newer is pretty much always better. You can either build from master, or grab packages from [the box64-debs repo](https://github.com/ryanfortner/box64-debs), but be aware that even just a day's worth of commits _will_ make a difference.

As for your WINE version, there seems to be some regressions going from 8.10 to 8.11, so it's your choice on whether to use either. 
You can get WINE 8.10 [here](https://github.com/Pi-Apps-Coders/files/releases/download/large-files/wine-8.10.tar.gz) and 8.11 [here](https://github.com/Pi-Apps-Coders/files/releases/download/large-files/wine-8.11.tar.gz). 
Keep in mind that some issues are present in WINE's WOW64 support and as such you should probably be testing on an AMD64 machine to compare functionality.

Update:
~~All Direct3D applications are broken in wine wow64 unless you have applied wine mr3053. They still don't function correctly in most cases, but crashes are reduced.~~ This has been merged as of wine 8.12.

The current blocker for direct3d applications is now that wined3d will pass back 64bit pointers from buffer maps directly back to direct3d applications, and thus most of their graphics will be broken. This can be fixed by wine mr3047, but the code path for platforms without MAP_32BIT (which aarch64 uses) is broken and as such this isn't a solution for now.

Tested applications:
Counter-Strike: Works intermittently as of 7689d0d with Wine 8.10
Counter-Strike Source: Runs as of 1f65d3a if wine mr3053 is applied but will not display anything other than a quit button. This is a [wine bug](https://bugs.winehq.org/show_bug.cgi?id=55045) and also occurs on x86.
RVGL; Runs correctly and FAST!!!!! as of 7817d62. Makes it to loading screen in 770cba7e, but had regressed to crashing right before it as of 7689d0d.
OpenMPT:  Opens but will not play audio, has been like this since at least 770cba7.
Deus Ex: crashes before menu, but plays audio, has been like this since at least 770cba7. Works if using the opengl render(can get to training), crashes in cutscenes, quite slow.
UT99: Various graphical errors but functions as of 1f65d3a. Use OpenGL renderer. Quite smooth actually.
Unreal Gold: Use OpenGL renderer, runs fine as of 1f65d3a.
Starsiege Tribes: Launches but controls doesn't work and won't read configs correctly, and no networking, as of 1f65d3a at least.

**Comment by ksco** [comment]
You might still want to create separate issues for each of the games. Creating an issue per game is convenient for the developer to track and fix them one by one. Mix'em together will make things complicated.

You can always link them back here if you want to track the progress in "one place". But all the info around an issue will be obsolete once that issue got fixed. As box64 WOW64 support is in alpha and it's fast envolving, these issues might get fixed just in days.

**Comment by ruje0504** [comment]
Prince of Qin、媚肉の香り(Biniku no Kaori )、biko4、河原崎家の一族2(Kawarazaki-ke no Ichizoku 2)、僕は天使じゃないよ(Boku wa Tenshi Janai yo) for normal

**Comment by AndreRH** [comment]
https://gitlab.winehq.org/wine/wine/-/merge_requests/3053 is merged

**Reply by ptitSeb** [reply]
Ah yeah, that unsupported persistent buffer error was breaking everything on wow64! That's some good news :)

**Comment by DocMAX** [comment]
wow64 is working pretty good! @ptitSeb thank you.
thats why i can use archlinux as os.
